### DISCUSSION TITLE
vCenter Custom Attributes as Netbox Tags

### DISCUSSION THREAD

**Discussion by amh-lilybloom** [bb-Ricardo/netbox-sync#312] [General]
Hello,

I haven't seen it mentioned as possible in the docs, but it doesn't seem like it would be complicated to implement:

I have a CustomAttribute called "Department" in vCenter.  I'd like to be able to match this field with a corresponding tag in Netbox.  For example a VM created for the department XYZ would get a corresponding Netbox tag XYZ - making it findable if you pull up the tag for that department (along with other Netbox devices/entries for this department).

Is this currently possible or a candidate for a feature request?

**Comment by bb-Ricardo** [comment]
Hi @amhwbloomingdale,

would it help to use a tag in vCenter:
https://github.com/bb-Ricardo/netbox-sync/blob/3089543f31960ebf48520d39c95d0e573ed6ac64/settings-example.ini#L283-L296

Then you could either tag the vm or the folder or the cluster and this tag, if defined in settings, would be added to the vm

**Reply by amh-lilybloom** [reply]
This does make sense.  I can run a PowerCLI script to convert the custom field to a tag.

Is there a means in the settings to filter what tags get sent over from the VM?  Similar to the filter on VM names via `vm_exclude_filter`

**Reply by bb-Ricardo** [reply]
all attached tags which the script finds will be synced. there is no exclude option. This would add at least three more config options.

**Reply by amh-lilybloom** [reply]
Fair enough.  I think the current setup is sufficient and I can handle any logic specific to my use case on my side

**Reply by bb-Ricardo** [reply]
sounds great. thank you

**Comment by MrgampRus** [comment]
Hello, I asked bb-Ricardo about similar functionality, maybe bb-Ricardo doesn't quite understand what we want.
https://github.com/bb-Ricardo/netbox-sync/discussions/293

**Reply by bb-Ricardo** [reply]
I do understand what you would like to achive. But asking to achive similar results with a working funcionallity first, should be ok.
